freqtrade/optimize/backtesting.py

```diff
@@ -920,7 +920,7 @@ def handle_left_open(self, open_trades: Dict[str, List[LocalTrade]],
                 trade.close(exit_row[OPEN_IDX], show_msg=False)
                 LocalTrade.close_bt_trade(trade)
 
-    def trade_slot_available(self, max_open_trades: int, open_trade_count: int) -> bool:
+    def trade_slot_available(self, max_open_trades: int | float, open_trade_count: int) -> bool:
```

Comment:
assuming this change is done to fix a mypy error ... (and wherever this type changed elsewhere ...)

whatever is triggering the fact that `max_open_trades` can be a float  - it's wrong and needs to be fixed "there".
you can't have 2.5 trades open - it's either 2 - or 3 - therefore must be an integer.

Reply:
I wanted to consider the `max_open_trades` type in the strategy as `int` and `float('inf')`, since when the config is parsed the `-1` is converted to `float('inf')`.

But if I understand well i should allow only for integer values for `max_open_trades` in the strategy (so that -1 == infinity), and so loading a strategy with no value for `max_open_trades` in the config and with `max_open_trades = -1` set in the strategy should result in a config with `max_open_trades = float('inf')`, and vice versa when loading strategy with a config that has max_open_trades == float('inf') I should set -1 in the strategy?